atlasdb-timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/PaxosTimeLockServerIntegrationTest.java

```diff
@@ -0,0 +1,170 @@
+import com.palantir.lock.LockDescriptor;
+import com.palantir.lock.LockMode;
+import com.palantir.lock.LockRefreshToken;
+import com.palantir.lock.LockRequest;
+import com.palantir.lock.RemoteLockService;
+import com.palantir.lock.StringLockDescriptor;
+import com.palantir.timestamp.TimestampService;
+
+import io.dropwizard.testing.ResourceHelpers;
+import io.dropwizard.testing.junit.DropwizardAppRule;
+
+public class PaxosTimeLockServerIntegrationTest {
+    private static final String NOT_FOUND_CODE = "404";
+
+    private static final String CLIENT_1 = "test";
+    private static final String CLIENT_2 = "test2";
+    private static final String NONEXISTENT_CLIENT = "nonexistent";
+    private static final String INVALID_CLIENT = "test2\b";
+
+    private static final Optional<SSLSocketFactory> NO_SSL = Optional.absent();
+    private static final String LOCK_CLIENT_NAME = "lock-client-name";
+    private static final SortedMap<LockDescriptor, LockMode> LOCK_MAP = ImmutableSortedMap.of(
+            StringLockDescriptor.of("lock1"), LockMode.WRITE);
+
+    @ClassRule
+    public static final DropwizardAppRule<TimeLockServerConfiguration> APP = new DropwizardAppRule<>(
+            TimeLockServer.class,
+            ResourceHelpers.resourceFilePath("paxosSingleServer.yml"));
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+        FileUtils.deleteDirectory(new File("var/test/data"));
+    }
+
+    @Test
+    public void lockServiceShouldAllowUsToTakeOutLocks() throws InterruptedException {
+        RemoteLockService lockService = getLockService(CLIENT_1);
+
+        LockRefreshToken token = lockService.lock(LOCK_CLIENT_NAME, LockRequest.builder(LOCK_MAP)
+                .doNotBlock()
+                .build());
+
+        assertThat(token).isNotNull();
+
+        lockService.unlock(token);
+    }
+
+    @Test
+    public void lockServiceShouldAllowUsToTakeOutSameLockInDifferentNamespaces() throws InterruptedException {
+        RemoteLockService lockService1 = getLockService(CLIENT_1);
+        RemoteLockService lockService2 = getLockService(CLIENT_2);
+
+        LockRefreshToken token1 = lockService1.lock(LOCK_CLIENT_NAME, LockRequest.builder(LOCK_MAP)
+                .doNotBlock()
+                .build());
+        LockRefreshToken token2 = lockService2.lock(LOCK_CLIENT_NAME, LockRequest.builder(LOCK_MAP)
+                .doNotBlock()
+                .build());
+
+        assertThat(token1).isNotNull();
+        assertThat(token2).isNotNull();
+
+        lockService1.unlock(token1);
+        lockService2.unlock(token2);
+    }
+
+    @Test
+    public void lockServiceShouldNotAllowUsToRefreshLocksFromDifferentNamespaces() throws InterruptedException {
+        RemoteLockService lockService1 = getLockService(CLIENT_1);
+        RemoteLockService lockService2 = getLockService(CLIENT_2);
+
+        LockRefreshToken token1 = lockService1.lock(LOCK_CLIENT_NAME, LockRequest.builder(LOCK_MAP)
```

Comment:
Hmm, did you mean on the next test? This one should stay as it is I think (since there are two tokens), but the next method has a single token that's named `token1`